kernel/aster-nix/src/fs/procfs/template/mod.rs

```diff
@@ -40,57 +44,87 @@ impl Common {
     }
 
     pub fn metadata(&self) -> Metadata {
-        *self.metadata.read()
+        *self.metadata.get()
     }
 
     pub fn ino(&self) -> u64 {
-        self.metadata.read().ino as _
+        self.metadata.get().ino as _
     }
 
     pub fn size(&self) -> usize {
-        self.metadata.read().size
+        self.metadata.get().size
     }
 
     pub fn atime(&self) -> Duration {
-        self.metadata.read().atime
+        self.metadata.get().atime
     }
 
     pub fn set_atime(&self, time: Duration) {
-        self.metadata.write().atime = time;
+        self.writer_lock.lock();
+        let reclaimer = self.metadata.replace({
```

Comment:
Sorry, this is not a good use case for RCU. Yes, using RCU reduces the overhead of the getter methods. But the setter methods gets much expensive. For file systems, updating the file attributes may be frequent. So making the setter methods significantly more expensive is highly undesirable.